### ISSUE TITLE
Dealing with Multi-Label Classifier Models

### ISSUE THREAD

**Issue by RamiAwar** [dichotomies/proxy-nca#5]
What if we had target values consisting of multiple labels?

then this would have to change: 
`Z_i = torch.index_select(
        proxies_normed,
        0,
        Variable(
            torch.LongTensor(
                np.reshape(
                    np.setdiff1d(
                        np.arange(0, nb_classes),
                        ts.data.cpu().numpy()[i]
                    ),
                    (nb_classes - 1)
                )
            )
        ).cuda()
    )` 

to something more dynamic, instead of forcing `(nb_classes - 1)`, although the same concept applies, taking all "non-applicable" labels. 

But then how would I deal with this: 
`    n_dist = torch.sum(torch.exp(-torch.sum(torch.pow(Z_i - xs[i], 2), dim=1)))`

Any help is appreciated! Still haven't fully understood all the code but I'm getting there!
